src/app/shared/services/course.service.ts

```diff
@@ -1,65 +1,56 @@
 import { Injectable } from '@angular/core';
-import { Course } from '../interfaces/course.interface.';
+import { HttpClient} from '@angular/common/http'
+import { BehaviorSubject, Observable, Subject } from 'rxjs';
+
+import { HttpCourse } from '../interfaces/http-course.interface';
 
 @Injectable({
   providedIn: 'root'
 })
 export class CourseService {
 
   idCounter = 3;
+  private quanityOfVisibleCourses = 3;
+  private apiUrl = 'http://localhost:3004';
+  inputSubject = new Subject<string>();
+  loadingSubject = new Subject<boolean>();
 
-  constructor() { }
+  constructor(private http: HttpClient) {
+    this.inputSubject = new BehaviorSubject<string>('');
```

Comment:
Try not to initialize the variables in the constructor
Furthermore, you have already initialized this variable - so you can just use the behavioursubject in the very beginning

Reply:
sure, forgot about it during refactoring